irohad/consensus/yac/storage/impl/yac_proposal_storage.cpp

```diff
@@ -105,25 +102,32 @@ namespace iroha {
       // --------| private api |--------
 
       bool YacProposalStorage::shouldInsert(const VoteMessage &msg) {
-        return checkProposalHash(msg.hash.proposal_hash)
-            and checkPeerUniqueness(msg);
+        return checkProposalHash(msg.hash.proposal_hash) and checkPeerUniqueness(msg);
       }
 
       bool YacProposalStorage::checkProposalHash(ProposalHash vote_hash) {
         return vote_hash == hash_;
       }
 
+        //TODO: it is not really clear what "peer uniqueness" means. Either a better name or comments
+        // are needed here.
       bool YacProposalStorage::checkPeerUniqueness(const VoteMessage &msg) {
+            //TODO: rather than having complex return statements like this, it is usually a better design decision
+            // to encapsulate some of the logic in other functions (that are then easily testable using unit tests;
```

Comment:
I am not sure that understanding of the method function is a very big deal in this case. Maybe not so typical of looking for a functional style of programming in C++? This method, in fact, contains only one `for` cycle in terms of imperative programming.